molecule/default/tasks/openshift_cancel_builds.yml

```diff
@@ -0,0 +1,168 @@
+- block:
+  - set_fact:
+      build_ns: "cancel-builds"
+      build_config: "ruby-centos0"
+      is_name: "ruby-is"
+
+  - name: Ensure namespace
+    kubernetes.core.k8s:
+      kind: Namespace
+      name: "{{ build_ns }}"
+
+  - name: Create ImageStream
+    community.okd.k8s:
+      namespace: "{{ build_ns }}"
+      definition:
+        apiVersion: image.openshift.io/v1
+        kind: ImageStream
+        metadata:
+          name: "{{ is_name }}"
+        spec:
+          lookupPolicy:
+            local: false
+          tags: []
+
+  - name: Create build configuration
+    community.okd.k8s:
+      namespace: "{{ build_ns }}"
+      definition:
+        kind: BuildConfig
+        apiVersion: build.openshift.io/v1
+        metadata:
+          name: "{{ build_config }}"
+        spec:
+          source: 
+            dockerfile: |
+              FROM openshift/ruby-22-centos7
+              RUN sleep 300s
+              USER build0
+          strategy:
+            type: Docker
+          output: 
+            to:
+              kind: "ImageStreamTag"
+              name: "{{ is_name }}:1.0.0"
+
+  - name: Start Build from Build configuration
+    community.okd.openshift_start_build:
+      namespace: "{{ build_ns }}"
+      build_config_name: "{{ build_config }}"
+    register: new_build
+
+  - name: Cancel Build from Build configuration
+    community.okd.openshift_cancel_build:
```

Comment:
It would be more natural to me if these modules managed resources. I would imagine more something like:

```
openshit_build:
    state: present
    namespace: blabla
    build_config_name :blabla
```

```
openshit_build:
    state: absent (alias cancel)
    namespace: blabla
    build_config_name :blabla
```

Reply:
agree with you I was thinking about the same, but I dont remember I did follow this design

Reply:
Updated